cpp/src/arrow/type.h

```diff
@@ -986,6 +987,7 @@ class ARROW_EXPORT Decimal128Type : public DecimalType {
 
   static constexpr int32_t kMinPrecision = 1;
   static constexpr int32_t kMaxPrecision = 38;
+  using c_type = arrow::Decimal128;
```

Comment:
I'm not convinced this is right. Why do you need this?

Reply:
To have the output buffer typed, first

using out_type = typename O::c_type;

then

auto out_data = output->GetMutableValues<out_type>(1);

Reply:
I changed the implementation and this is not needed anymore.